README.md

```diff
@@ -71,45 +71,46 @@ headers = ApiAuth.headers(["Content-Type": "application/json"], "/post/path",
 
 To authenticate all requests for a particular pipeline, create a new
 plug and configure it to use `ApiAuth`.
-Note that you have to add it to `endpoint.ex` and not `router.ex` because it's
-[not always possible](https://github.com/phoenixframework/phoenix/issues/459)
-to get the body of a request in a regular pipeline:
 
-```elixir
-# lib/myapp_web/endpoint.ex
-
-defmodule Myapp.Endpoint do
-  use Phoenix.Endpoint, otp_app: :myapp
+Note that `Plug.Conn.read_body/2` can only be called once. This means that
+if you need the body for something else, you have to make sure to save it.
+There are also particular issues with JSON APIs due to the way `Plug.Parsers.JSON`
+works.
 
-  ...
+[This issue](https://github.com/phoenixframework/phoenix/issues/459)
+has some discussion about these problems and different workarounds.
+The sample code below assumes that the raw body has been saved in `conn.assigns.raw_body`.
 
-  # Add the `Authentication` plug immediately before `Plug.Parsers`.
-  # What `mount: "api"` does is limit the routes that are authenticated.
-  # In this example, only routes that start with `api/` are authenticated.
-  plug Myapp.Plugs.Authentication, mount: "api"
+```elixir
+# lib/myapp_web/router.ex
 
-  plug Plug.Parsers,
-    parsers: [:urlencoded, :multipart, :json],
-    pass: ["*/*"],
-    json_decoder: Poison
+defmodule MyappWeb.Router do
+  use MyappWeb, :router
 
-  ...
+  pipeline :api do
+    plug(Myapp.AuthenticationPlug)
+  end
 end
 ```
 
 ```elixir
-# lib/myapp_web/plugs/authentication.ex
+# lib/myapp_web/plugs/authentication_plug.ex
+
+defmodule MyappWeb.AuthenticationPlug do
+  @moduledoc """
+  Authentication plug
+  Using the `api_auth` package (https://github.com/TheGnarCo/api_auth_ex#phoenix)
+  this plug allows requests to continue through the pipeline only if they
+  have a valid HMAC signature.
+  """
 
-defmodule Myapp.Plugs.Authentication do
   import Plug.Conn
 
   def init(default), do: default
 
-  def call(conn, [mount: mount]) do
-    case conn.path_info do
-      [^mount | _] -> authorize(conn)
-      _            -> conn
-    end
+  def call(conn, _default) do
+    conn
+    |> authorize()
```

Comment:
From @BlakeWilliams in #37, in regards to the implementation of `authorize()`:

> I was thinking of creating a URI struct then using it to generate the path for us.

> eg:

`URI.parse(full_path) |> Map.put(:query, query_string) |> URI.to_string()`
> Might be overkill in this case, but it handles the conditional for us and maybe some other cases.

Reply:
This looks like it'll work, replacing `full_path` with `request_path` and removing the need for the `full_path` variable entirely.